spec/controllers/users/backup_code_setup_controller_spec.rb

```diff
@@ -1,11 +1,33 @@
 require 'rails_helper'
 
 describe Users::BackupCodeSetupController do
-  it 'creates backup codes' do
+  it 'creates backup codes and logs expected events' do
     user = build(:user, :signed_up)
+    attributes_visited = {
+      success: true,
+      errors: {},
+      mfa_method_counts: {phone: 1},
+      pii_like_keypaths: [[:mfa_method_counts, :phone]],
+      error_details: nil,
+      enabled_mfa_methods_count: 1,
+    }
+    attributes_created = {
+      enabled_mfa_methods_count: 2,
+    }
+    
     stub_sign_in(user)
+    stub_analytics
+    stub_attempts_tracker
+
     expect(PushNotification::HttpPush).to receive(:deliver).
       with(PushNotification::RecoveryInformationChangedEvent.new(user: user))
+    expect(@analytics).to receive(:track_event).
+      with('Backup Code Setup Visited', attributes_visited)
+    expect(@analytics).to receive(:track_event).
+      with('Backup Code Created', attributes_created)
```

Comment:
inlining attributes

```suggestion
    expect(@analytics).to receive(:track_event).
      with('Backup Code Setup Visited', {
        success: true,
        errors: {},
        mfa_method_counts: { phone: 1 },
        pii_like_keypaths: [[:mfa_method_counts, :phone]],
        error_details: nil,
        enabled_mfa_methods_count: 1,
    )
    expect(@analytics).to receive(:track_event).
      with('Backup Code Created', enabled_mfa_methods_count: 2)
```

Reply:
Done!  Thanks for suggesting the streamline!